bindings/capacitorjs/src/lib/WalletApi.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { AccountId, AccountSyncOptions, CreateAccountPayload } from '../types/account';
+import { AccountManagerOptions } from '../types';
+import { Account } from './Account';
+import { AccountManager } from './AccountManager';
+
+type ProfileManager = Awaited<ReturnType<typeof AccountManager>>
+type RecoverAccounts = Parameters<ProfileManager['recoverAccounts']>
+interface RecoverAccountsPayload {
+    accountStartIndex: number, 
+    accountGapLimit: number, 
+    addressGapLimit: number, 
+    syncOptions?: AccountSyncOptions
+}
+interface ProfileManagers {
+    [key: string]: ProfileManager
+}
+const profileManagers: ProfileManagers = {}
+
+const WalletApi = {
+    async createAccountManager(id: AccountId, options: AccountManagerOptions) {
+        const manager = await AccountManager(options)
+        manager.id = id
+        profileManagers[id] = manager
+        return manager
+    },
+    async createAccount(managerId: AccountId, payload: CreateAccountPayload) {
+        const manager = profileManagers[managerId]
+        const account = await manager.createAccount(payload)
+        bindMethodsAcrossContextBridge(Account.prototype, account)
+        return account
+    },
+    deleteAccountManager(managerId: AccountId) {
+        if (managerId && managerId in profileManagers) {
+            delete profileManagers[managerId]
+        }
+    },
+    async getAccount(managerId: AccountId, index: number) {
+        const manager = profileManagers[managerId]
+        const account = await manager.getAccount(index)
+        bindMethodsAcrossContextBridge(Account.prototype, account)
+        return account
+    },
+    async getAccounts(managerId: AccountId) {
+        const manager = profileManagers[managerId]
+        const accounts = await manager.getAccounts()
+        accounts.forEach((account) => bindMethodsAcrossContextBridge(Account.prototype, account))
+        return accounts
+    },
+    async recoverAccounts(managerId: AccountId, payload: RecoverAccountsPayload) {
+        const manager = profileManagers[managerId]
+        const accounts = await manager.recoverAccounts(...Object.values(payload) as RecoverAccounts)
+        accounts.forEach((account) => bindMethodsAcrossContextBridge(Account.prototype, account))
+        return accounts
+    },
+}
+
+function bindMethodsAcrossContextBridge(prototype, object) {
+    const prototypeProperties = Object.getOwnPropertyNames(prototype)
+    prototypeProperties.forEach((key) => {
+        if (key !== 'constructor') {
+            object[key] = object[key].bind(object)
+        }
+    })
+}
+
+export { WalletApi };
```

Comment:
Interesting approach to put the `bindMethodsAcrossContextBridge` function here. This is electron specific, right?

Reply:
As discussed in the call, this is very specific to Electron, so it might not be needed for the capacitor version. https://www.electronjs.org/docs/latest/api/context-bridge#api-functions

Reply:
yes I remember when you've mentioned it, but I really wasn't sure how that internally worked in that moment. The issue is that without calling that function `Account` class methods remains in the prototype.
Then of review it well, now I think that the "API Functions" paragraph at your link uses the "bind" word but does not mention it's needed to use the `Function.prototype.bind()` method (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind) to expose the class methods or "bind" them through the contextBridge.

So I think it's actually related to the wrong use of the class constructor, where there is returned a non-primitive value (currently is returned a promise object, resolved as another object with properties and methods), becoming the value of the `new` expression, and the value of `this` end up dropped. Refs:
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_Classes#constructor
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/new#description
IMO that's the why in that case it's used the `bindMethodsAcrossContextBridge` function, it's not an Electron thing.

I think is not a good idea to use classes in JavaScript when are needed return promises in the constructor rather than primitive values, that's the why I changed by factory functions. I testing now the new changes, it's taking a bit longer 😅

Reply:
After checking this out, we agreed that we would replace Account and AccountManager classes with the `createAccount` and `createAccountManager` factory functions. Additionally, we need some types declared as well, but this might come in the next PR.

Reply:
I've just updated the files 👌